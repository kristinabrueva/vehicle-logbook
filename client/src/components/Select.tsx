interface SelectProps {
  name: string;
  required?: boolean;
  value: string;
  options: string[];
  type?: "button" | "submit" | "reset" | undefined;
  handleChange?: (e: any) => void;
}

const Select = ({
  name,
  required = true,
  value,
  options,
  handleChange,
}: SelectProps): JSX.Element => (
  <select
    id={name}
    name={name}
    value={value}
    required={required}
    onChange={handleChange}
    className="block border-2 max-w-[160px] text-left text-base"
  >
    <option disabled={true} value="">
      {name}
    </option>
    {options.map((i) => (
      <option key={i} value={i}>
        {i}
      </option>
    ))}
  </select>
);

export default Select;
