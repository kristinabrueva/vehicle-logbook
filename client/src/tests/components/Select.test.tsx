import React from "react";
import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Select from "../../components/Select";

describe("Tests Select", () => {
  it("renders select correctly", () => {
    const { container, getAllByTestId, getAllByText } = render(
      <Select
        name="fruits"
        value="Apple"
        options={["Apple", "Pear", "Orange"]}
        testId="testSelect"
      />
    );
    expect(container).toMatchSnapshot();
    expect(getAllByTestId("testSelect")).toHaveLength(1);
    expect(getAllByText("fruits")).toHaveLength(1);
  });

  it("simulates selection", () => {
    const fruits = ["Apple", "Pear", "Orange"];
    const { getByTestId, getByText, getByRole } = render(
      <Select
        name="fruits"
        value="Apple"
        options={fruits}
        testId="testSelect"
      />
    );
    userEvent.selectOptions(
      getByTestId("testSelect"),
      getByRole("option", { name: "Apple" })
    );
    expect(
      (getByRole("option", { name: "Apple" }) as HTMLOptionElement).selected
    ).toBe(true);
    expect(
      (getByRole("option", { name: "Pear" }) as HTMLOptionElement).selected
    ).toBe(false);
  });
});
