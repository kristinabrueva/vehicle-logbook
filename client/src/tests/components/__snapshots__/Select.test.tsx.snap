// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Select renders select correctly 1`] = `
<div>
  <select
    class="block border-2 max-w-[160px] text-left text-base"
    data-testid="testSelect"
    id="fruits"
    name="fruits"
    required=""
  >
    <option
      disabled=""
      value=""
    >
      fruits
    </option>
    <option
      value="Apple"
    >
      Apple
    </option>
    <option
      value="Pear"
    >
      Pear
    </option>
    <option
      value="Orange"
    >
      Orange
    </option>
  </select>
</div>
`;
